multivac/gather_job_data.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+import glob
+import json
+import os
+import sys
+import re
+
+PROJECT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+sys.path.append(PROJECT_DIR)
+
+
+class GatherData:
+    def __init__(self):
+        self.workflow_run_jobs_dir = 'workflow_run_jobs'
+        self.output_dir = 'gathered_data'
+        self.gathered_data = dict()
+
+    def gather_data(self):
+        for jobs in glob.glob(os.path.join(self.workflow_run_jobs_dir,
+                                           '*.json')):
+            runner_version = None
+            # Load info about jobs from .json
+            with open(jobs, 'r') as f:
+                job = json.load(f)
+
+            if job['conclusion'] == 'skipped':
+                continue
+            if 'aarch64' in job['name']:
+                platform = 'aarch64'
+            else:
+                platform = 'amd64'
+
+            # Load info about jobs from .log, if there are logs
+            logs = jobs.rstrip('json') + 'log'
+            with open(logs, 'r') as f:
+                time_queued = f.readline()[0:19] + 'Z'
+                for _ in range(10):
+                    line = f.readline()
+                    if 'runner version' in line:
+                        runner_version = re.findall(r'[0-9]+\.[0-9]+\.[0-9]+',
```

Comment:
It works most optimal if you compile the regexp first, https://docs.python.org/3/howto/regex.html#compiling-regular-expressions

here's an example
```ipython
In [21]: strings
Out[21]:
['',
 '2022-03-03T22:35:48.7491360Z Requested labels: ubuntu-18.04',
 '2022-03-03T22:35:48.7491411Z Job defined at: tarantool/tarantool/.github/workflows/fuzzing.yml@refs/pull/6879/merge',
 '2022-03-03T22:35:48.7491435Z Waiting for a runner to pick up this job...',
 '2022-03-03T22:36:00.8387968Z Job is waiting for a hosted runner to come online.',
 '2022-03-03T22:36:06.9732725Z Job is about to start running on the hosted runner: GitHub Actions 2 (hosted)',
 "2022-03-03T22:36:16.7102584Z Current runner version: '2.287.1'",
 '2022-03-03T22:36:16.7135077Z ##[group]Operating System',
 '2022-03-03T22:36:16.7135680Z Ubuntu',
 '2022-03-03T22:36:16.7136032Z 18.04.6',
 '2022-03-03T22:36:16.7136742Z LTS',
 '2022-03-03T22:36:16.7137048Z ##[endgroup]',
 '2022-03-03T22:36:16.7137542Z ##[group]Virtual Environment',
 '']

In [22]: p = re.compile("Current runner version: '(\d*.\d*.\d*)'")

In [23]: for string in strings:
    ...:     s = p.search(string)
    ...:     if s:
    ...:         print (s.group(0))

Current runner version: '2.287.1'
```